packages/vite-plugin-svelte/src/utils/compile.ts

```diff
@@ -6,6 +6,8 @@ import { SvelteRequest } from './id';
 import { safeBase64Hash } from './hash';
 import { log } from './log';
 
+const scriptLangRE = /<script.+lang="(.+)".*>/;
```

Comment:
```suggestion
const scriptLangRE = /<script [^>]*lang=[\"']?([^\"' >]+)[\"']?[^>]*>/;
```

avoids greed and allows all quote styles:
```
<script lang="ts" foo="bar">
<script lang='ts'>
<script lang=ts>
<script lang='ts' foo='bar'>
<script lang=ts foo=bar>
```

one weakness is it would not match a quoted lang value that contains a space `lang="why do that"`

Reply:
Always great to be more robust 👍